include/inl/gnc_containers.inl

```diff
@@ -0,0 +1,64 @@
+//
+// include/inl/gnc_containers.inl
+// PSim
+//
+// Contributors:
+//   Kyle Krol  kpk63@cornell.edu
+//
+// Pathfinder for Autonomous Navigation
+// Space Systems Design Studio
+// Cornell Univeristy
+//
+
+#include "../gnc_containers.hpp"
+
+namespace gnc {
+
+template <typename T, std::size_t N>
+constexpr CircularBuffer<T, N>::CircularBuffer()
+: array(), s(0), n(0) { }
+
+template <typename T, std::size_t N>
+constexpr bool CircularBuffer<T, N>::full() const {
+  return size() == max_size();
+}
+
+template <typename T, std::size_t N>
+constexpr bool CircularBuffer<T, N>::empty() const {
+  return size() == 0;
+}
+
+template <typename T, std::size_t N>
+constexpr std::size_t CircularBuffer<T, N>::size() const {
+  return n;
+}
+
+template <typename T, std::size_t N>
+constexpr std::size_t CircularBuffer<T, N>::max_size() const {
+  return N;
+}
+
+template <typename T, std::size_t N>
+constexpr T const &CircularBuffer<T, N>::operator[](std::size_t i) const {
```

Comment:
This implementation is missing a pop() function. It's not necessary for your application but just thought I'd point that out as a generally needed feature of ring buffers (we definitely need it on the flight software side)

Reply:
Ah okay yeah I could add that in pretty easily -> just bump up the `s` member variable by one. `peek` and `pop` work?

Reply:
Sure! Also we don't need ring buffers on FSW anymore, so it's a totally optional feature

Reply:
Ah okay gotcha - probs just leave it for now then